### PR TITLE
[docs][yba] Control Plane RBAC with custom roles (MVP)

### PR DESCRIPTION
**Pull request by ddhodge**
Add custom Roles functionality.
DOC-127

@netlify /preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac/

### CODE REVIEW

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/rbac-platform.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+* **Super Admin** is the first user that is created during the tenant registration. This role has the highest level of privilege that allows all read and write actions on all YugabyteDB Anywhere resources. There can be only one Super Admin in a tenant. This Super Admin can perform the following:
+
+  * Manage all resources, including universes, nodes, backup, restore, and cloud providers.
+  * Manage the user access control by creating and managing users.
+
+  For more information, see [admin user](../../configure-yugabyte-platform/create-admin-user/).
+
+* **Admin** has privileges that are similar to the Super Admin, except that the Admin cannot manage the global scope artifacts and actions, such as runtime configuration settings and LDAP authentication.
+
+* **Backup Admin** has access the backup related tasks, such as the following:
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/rbac-platform.md:55]:
```suggestion
* **Backup Admin** has access to the backup related tasks, such as the following:
```

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/rbac-platform.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+Users are assigned roles, which determine what they actions they can perform.
+
+### Create, modify, and delete users
+
+As a Super Admin or Admin, you can invite new users and manage existing users for your tenant.
+
+You can invite new users to your tenant as follows:
+
+* Navigate to **Admin > User Management > Users** and click **Add User**.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/rbac-platform.md:28]:
```suggestion
* Navigate to **Admin > User Management > Users**, and click **Add User**.
```

File: docs/content/preview/yugabyte-platform/security/authorization-platform.md [hunk 3]
```diff
@@ -11,42 +11,52 @@ menu:
...
+--------------+------------------------------------------------------------+-----------
+ postgres     | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
+ yb_db_admin  | No inheritance, Cannot login                               | {}
+ yb_extension | Cannot login                                               | {}
+ yb_fdw       | Cannot login                                               | {}
+ yugabyte     | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
+```
 
-## Create, modify, and delete users
+The following table describes the default YSQL roles and users in YugabyteDB Managed clusters.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/security/authorization-platform.md:45]: Why do we want to mention about YSQL roles pertaining to YBM in YBA docs?

File: docs/content/preview/yugabyte-platform/security/authorization-platform.md [hunk 4]
```diff
@@ -11,42 +11,52 @@ menu:
...
 
-A role is a set of predefined permissions in YugabyteDB Anywhere. The following roles are available:
+The YugabyteDB database on your universe includes a set of default users and roles in YSQL and YCQL.
 
-* **Super Admin** is the first user that is created during the tenant registration. This role has the highest level of privilege that allows all read and write actions on all YugabyteDB Anywhere resources. There can be only one Super Admin in a tenant. This Super Admin can perform the following:
+## YSQL default roles and users
 
-  * Manage all resources, including universes, nodes, backup, restore, and cloud providers.
-  * Manage the user access control by creating and managing users.
+To view the roles in your cluster, enter the following command:
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/security/authorization-platform.md:28]:
```suggestion
To view the roles in your universe, enter the following command:
```

File: docs/content/preview/yugabyte-platform/security/security-checklist-yp.md [hunk 5]
```diff
@@ -11,27 +11,30 @@ menu:
...
 
 ## Role-based access control
 
-Roles can be assigned to grant users only the essential privileges based on the operations they need to perform against YugabyteDB Anywhere. Typically, a Super Admin role is created first. The Super Admin can create additional admins and other users with fewer privileges.
+Roles can be assigned to grant users only the essential privileges based on the operations they need to perform in YugabyteDB Anywhere, and in YugabyteDB universes. Typically, a Super Admin role is created first. The Super Admin can create additional admins and other users with fewer privileges.
+
+For information on how to manage YugabyteDB Anywhere users and roles, see [Manage YugabyteDB Anywhere users](../../administer-yugabyte-platform/rbac-platform/).
 
-For information on how to enable role-based access control in YugabyteDB Anywhere, see [Authorization](../authorization-platform).
+For information on how to manage roles and users in YugabyteDB universes, see [Database authorization](../authorization-platform).
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/security/security-checklist-yp.md:34]:
```suggestion
For information on how to manage roles and users in YugabyteDB Anywhere universes, see [Database authorization](../authorization-platform).
```

**Review by aishwarya24 — COMMENTED** [review]
some comments

**Comment by ddhodge** [comment]
@haikarthikssk @Arpit-yb Can you review or suggest someone to review

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+type: docs
+---
+
+YugabyteDB Anywhere uses a role-based access control (RBAC) model to manage access to your YugabyteDB Anywhere instance (whether via the UI or the REST API). Using roles, you can enforce the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) (PoLP) by ensuring that users have the precise permissions needed to fulfill their roles while mitigating the risk of unauthorized access or accidental breaches. A role defines a set of permissions that determine what features can be accessed by account users who have been assigned that role.
+
+## Users
+
+A user can interact with a YugabyteDB Anywhere through the UI or REST API.
+
+Users are assigned roles, which determine what they actions they can perform.
```
Comment (vpatibandla-yb) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:21]: which determine what are the actions they can perform.

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+* **Read-only** access level provides viewer permission to the UI and API.
+
+### Create, modify, and delete roles
+
+As a Super Admin or Admin, you can create and modify roles.
+
+To create a custom role, do the following:
+
+1. Navigate to **Admin > User Management > Roles**, then click **Add Role** to display the **Create a Role** dialog.
```
Comment (vpatibandla-yb) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:74]: Admin > Access Management > Roles

Comment (vpatibandla-yb) [reply docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:74]: With the new RBAC, we can not only create Custom Roles but also scope the customer roles at the universe level.

Please also include the flow to modify the role bindings of the users from the Admin > Access Management > Users > Edit Assigned Roles option for the user.

Comment (vpatibandla-yb) [reply docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:74]: The new RBAC feature which allows custom roles can be turned on by using the runtime flag yb.rbac.use_new_authz. This is an internal Runtime config as of now. I will make the change to PUBLIC once I get the QA approval.

The runtime config can be edited from the UI from the Admin > Advanced > Global Configuration section and the runtime config key yb.rbac.use_new_authz should be set to true for the user to use the new RBAC logic. Once the runtime config is set to true, it cannot be turned off

Comment (vpatibandla-yb) [reply docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:74]: Please write a section on how to assign universes for Custom roles

Comment (vpatibandla-yb) [reply docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:74]: Phase 1 MVP: The resource that we deal with are Universes. Later phases will deal with other kind of resources

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+1. Enter a name for the role.
+1. Enter a description for the role.
+1. Click **Select Permissions**.
+1. Select the permissions to assign to the role and click **Select** when you are done.
+1. Click **Save**.
+
+To modify a user role or delete a role, navigate to **Admin > User Management > Roles**. Click **Actions** that corresponds to the specific role, and then select either **Edit Role** or **Delete Role**.
```
Comment (vpatibandla-yb) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:81]: This is the old flow, do we want to include the new RBAC workflows too?

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md (hunk 6 above)
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:21]:
```suggestion
Users are assigned roles, which define the set of actions users can perform.
```

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md (hunk 8 above)
Comment (ddhodge) [reply docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:81]: We want only the new flow I would have thought

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+* **Backup Admin** has access to the backup related tasks, such as the following:
+
+  * Manage database backups and restore operations.
+  * Create new backups.
+  * Delete any existing backup or backup schedule.
+  * Edit existing backups.
+  * Read-only permissions for all other resources in YugabyteDB Anywhere.
+
+* **Read-only** access level provides viewer permission to the UI and API.
```
Comment (Sahith02) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:83]: Theres another built-in role called "Connect Only".

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+
+Users are assigned [roles](#roles), which define the set of actions users can perform. You can also define a set of universes to which the user has access.
+
+### Create, modify, and delete users
+
+As a Super Admin or Admin, you can invite new users and manage existing users for your YugabyteDB Anywhere instance.
+
+You can assign built-in and custom roles to users to determine the actions they are allowed to perform, and specify the universes that they can access.
+
+To create a user, do the following:
```
Comment (Sahith02) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:29]: Would it be better to add images here to help guide the user better?

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+To edit a custom role, do the following:
+
+1. Navigate to **Admin > Access Management > Roles**, click **Actions** for the role to edit, and choose **Edit Role**.
+1. On the **Configurations** tab click **Edit Permissions**.
+1. Select the permissions to assign to the role and click **Confirm** when you are done.
+1. Click **Save**.
+
+To delete a role, navigate to **Admin > Access Management > Roles**, click **Actions** for the role to edit, and choose **Edit Role**.
+
+To view the users that have been assigned a role, navigate to **Admin > Access Management > Roles**, click **Actions** for the role, and choose **View Assigned Users**.
```
Comment (Sahith02) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md:122]: @ddhodge @vpatibandla-yb Should we also add here some of the limitations (from our internal design doc) that we have in RBAC phase 1, which will be solved in phase 2?

**Review by Sahith02 — APPROVED** [review]

**Pull request merged**